lib/service/Service.ts

```diff
@@ -195,36 +195,10 @@ class Service extends EventEmitter {
       return 'Taker\'s LND is not connected';
     }
 
-    body =  {
-      takerCoin,
-      makerCoin,
-      takerPubKey,
-      takerDealId: randomBytes(32).toString('hex'),
-      takerAmount: payload.receivingAmount,
-      makerAmount: payload.sendingAmount,
-    };
-
-    const deal: SwapDeal = {
-      myRole : SwapDealRole.Taker,
-      takerAmount : body.takerAmount,
-      takerCoin : body.takerCoin,
-      takerPubKey: body.takerPubKey,
-      takerDealId: body.takerDealId,
-      makerAmount: body.makerAmount,
-      makerCoin: body.makerCoin,
-      createTime: Date.now(),
-    };
-    this.pool.swapDeals.add(deal);
-    this.logger.debug(' swap deal: ' + JSON.stringify(deal));
-
-    this.logger.debug('sending to peer ' + payload.nodePubKey + ': ' + JSON.stringify(body));
-    const packet = new packets.DealRequest(body);
+    const peer = this.pool.getPeer(payload.nodePubKey);
+    this.swaps.beginSwap(takerCurrency, makerCurrency, takerPubKey, payload.receivingAmount, payload.sendingAmount, peer);
```

Comment:
It might be better for `Swaps` should contain `Pool` (same as `OrderBook`), instead of the opposite. Here it's just about having it query the peer by itself, but it could make other things easier as well (direct packet listening, etc.). It will create a unidirectional flow, instead of `Pool` using `Swaps` as a util.

Also, `Swaps` should probably be inside `OrderBook` (or `MatchesProcessor`, in case it will still be needed), to activate swaps.

Reply:
Sounds good and makes sense as well.